src/test/java/edu/ucsb/courses/controllers/ScheduleItemControllerTests.java

```diff
@@ -0,0 +1,487 @@
+        String url = String.format(urlTemplate, "1", "Adv App Programming", "CS 156");
+
+        DecodedJWT jwt = JWT.decode(userToken.substring(7));
+
+        ScheduleItem schedule = new ScheduleItem(1L,"CS 156", "Adv App Programming", jwt.getSubject(), 1L);
+
+        when(scheduleItemRepository.save(any(ScheduleItem.class)))
+                .thenReturn(schedule);
+
+        MvcResult response = mockMvc.perform(get(url).contentType("application/json").header(HttpHeaders.AUTHORIZATION, userToken)).andExpect(status().isOk())
+                .andReturn();
+
+        String responseString = response.getResponse().getContentAsString();
+        ScheduleItem returnVal = objectMapper.readValue(responseString, ScheduleItem.class);
+
+        assertEquals(returnVal, schedule);
+    }
+
+    @Test
+    public void test_deleteScheduleItemSuccess() throws Exception {
+        String expectedResult = "";
+        String urlTemplate = "/api/public/removeScheduleItem?id=%s";
+        String url = String.format(urlTemplate, "1");
+
+        DecodedJWT jwt = JWT.decode(userToken.substring(7));
+        ScheduleItem scheduleItem = new ScheduleItem(1L,"CS 156", "Adv App Programming", jwt.getSubject(), 1L);
+        Optional<ScheduleItem> opt = Optional.of(scheduleItem);
+
+        when(scheduleItemRepository.findById(any(Long.class)))
+                .thenReturn(opt);
+
+        MvcResult response = mockMvc.perform(delete(url).with(csrf()).contentType("application/json").header(HttpHeaders.AUTHORIZATION,userToken)).andExpect(status().isOk())
+                .andReturn();
+
+        String responseString = response.getResponse().getContentAsString();
+
+        assertEquals(expectedResult, responseString);
+    }
+
+    @Test
+    public void test_removeScheduleFailureNoAuth() throws Exception {
+        String expectedResult = "";
+        String urlTemplate = "/api/public/removeScheduleItem?id=%s";
+        String url = String.format(urlTemplate, "1");
+
+        //DecodedJWT jwt = JWT.decode(userToken.substring(7));
+
+        ScheduleItem scheduleItem = new ScheduleItem(1L,"CS 156", "Adv App Programming", "", 1L);
+
+        Optional<ScheduleItem> opt = Optional.of(scheduleItem);
+
+        when(scheduleItemRepository.findById(any(Long.class)))
+                .thenReturn(opt);
+
+        MvcResult response = mockMvc.perform(delete(url).with(csrf()).contentType("application/json").header(HttpHeaders.AUTHORIZATION,userToken)).andExpect(status().isBadRequest())
+                .andReturn();
+
+        String responseString = response.getResponse().getContentAsString();
+
+        assertEquals(expectedResult, responseString);
+    }
+
+    @Test
+    public void test_removeScheduleFailureNotFound() throws Exception {
+        String expectedResult = "";
+        String urlTemplate = "/api/public/removeScheduleItem?id=%s";
+        String url = String.format(urlTemplate, "1");
+
+        //DecodedJWT jwt = JWT.decode(userToken.substring(7));
```

Comment:
We can remove unused code instead of commenting it out.

Reply:
Yes